models/pipeline.js

```diff
@@ -85,7 +89,7 @@ module.exports = {
      * @property keys
      * @type {Array}
      */
-    keys: ['scmUrl'],
+    keys: ['scmRepo.id'],
```

Comment:
Don't change the key yet, that way it won't be backwards incompatible.  We can slowly migrate towards using it.